flytekit/extras/tasks/shell.py

```diff
@@ -192,26 +199,54 @@ def execute(self, **kwargs) -> typing.Any:
         Executes the given script by substituting the inputs and outputs and extracts the outputs from the filesystem
         """
         logger.info(f"Running shell script as type {self.task_type}")
+        breakpoint()
+
+        # somewhat hidden behavior and hacky at the moment, if script_args is in kwargs, grab it and save it
+        script_args = kwargs.pop("script_args") if "script_args" in kwargs else None
```

Comment:
feels hacky

Reply:
Here is a less hacky way of doing this (I think)
Instead of passing all kwargs to interpolate (which is why you need to pop here), use a list comp to grab the kwargs that match the specification defined in the inputs variable passed to __init__. Then you can leave anything in kwargs, and even just add script_args to the function definition so it isn't included in kwargs (probably)... does that make sense?

Reply:
Ditto for output_locs

Reply:
which function definition? You mean `execute()`? We cannot add any args to execute since we do not call it, we can only add key args that we specify in `inputs`. Basically, all variables in `inputs` are put into hte kwargs dict regardless of what we intend to use them for